### PR TITLE
feat(Core/CFBG): Added support module mod-cfbg

### PR DESCRIPTION
**Pull request by Winfidonarleyan**
#### Really? - Yes
- Finally, PR, which adds support for the module [mod-cfbg](https://github.com/Winfidonarleyan/mod-cfbg)
(A little humor, in connection with the recent events 👍 )

##### CHANGES PROPOSED:
-  Added support `mod-cfbg` module

##### TESTS PERFORMED:
- [x] Build win (VS 16.1.3)
- [x] Test in-game Random BG
- [x] Test in-game WSG

##### HOW TO TEST THE CHANGES:
1. Apply this PR
2. Added module [mod-cfbg](https://github.com/Winfidonarleyan/mod-cfbg)
3. Enable system

##### KNOWN ISSUES AND TODO LIST:
- [x] Test all BG + queue
- [x] Test all Arena + queue
- [x] Fix cpam chat
- [x] Fix language for fake players

##### Target branch(es): Master

### CODE REVIEW

**Comment by solidmaxtor** [comment]
add pr and mod-cfbg but all error in compilation for me.

**Comment by Winfidonarleyan** [comment]
@solidmaxtor I don't understand your message.

**Comment by solidmaxtor** [comment]
use git clone comand direct, but all error is repeat in compile for me.

File: apps/ci/ci-install-modules.sh [hunk 1]
```diff
@@ -9,3 +9,4 @@ git clone --depth=1 --branch=master https://github.com/azerothcore/mod-transmog.
 git clone --depth=1 --branch=master https://github.com/azerothcore/mod-npc-beastmaster.git modules/mod-npc-beastmaster
 git clone --depth=1 --branch=master https://github.com/azerothcore/mod-duel-reset.git modules/mod-duel-reset
 git clone --depth=1 --branch=master https://github.com/azerothcore/mod-premium modules/mod-premium
+git clone --depth=1 --branch=master https://github.com/Winfidonarleyan/mod-cfbg modules/mod-cfbg
```
Comment (FrancescoBorzi) [inline apps/ci/ci-install-modules.sh]: shall we use `https://github.com/azerothcore/mod-cfbg` instead ? I've just forked your repo. We usually put under `azerothcore/` the most important modules (and I believe your module is very important!)

Comment (Winfidonarleyan) [reply apps/ci/ci-install-modules.sh]: I change this, for only test pr with module.
I think we can do that.

**Comment by solidmaxtor** [comment]
Test again:
Same error i am test PR 2064 only and pr2010+2064, with and without  mod-cfbg, all same error.
(i am check  ./acore.sh  to install mod-cgbf work perfect)

Maybe this image help for check what is error or what am I failing XD

image

**Comment by Winfidonarleyan** [comment]
Need to resolve conflicts

**Comment by FrancescoBorzi** [comment]
@solidmaxtor you are using a branch that is not clean. Copy paste this command (this will also **reset** your sources, any unsaved work will be lost):

```
git checkout master; git reset --hard; git pull;
```

Now you are in a clean `master` branch, you can follow again the [procedure to test PRs](http://www.azerothcore.org/wiki/How-to-test-a-PR) and test this one.

**Comment by solidmaxtor** [comment]
Last compile all pass ok
I will add in main server por active test.

**Comment by solidmaxtor** [comment]
2 days test, no crashes, no reports for me.
Is possible config this for specific leves?  10 to 79 maybe for no affect level 80 bg?

**Comment by Winfidonarleyan** [comment]
@solidmaxtor, The main thing for us is to test the support of the module, then we can redo the module as we want

**Comment by Winfidonarleyan** [comment]
And, need test game chat

**Comment by wowmane** [comment]
@Winfidonarleyan 

I Will To full tested your commit But please next update:
https://github.com/azerothcore/azerothcore-wotlk/pull/2014

AZEROTHCORE NEED TO ANTICHEAT !

**Comment by Winfidonarleyan** [comment]
@wowmane, Hello, ahah, maybe maybe

**Comment by wowmane** [comment]
For me Worked.

**Comment by FrancescoBorzi** [comment]
Excellent @Winfidonarleyan and thanks a lot @solidmaxtor and @wowmane for testing it

**Comment by solidmaxtor** [comment]
Hello, i am test in main server now, report many chat errors inside cross BG for Messengers in code or addons spam codes.

And have xtrain chat mensajes in all server.

CFBG codes

CFBG mensajes extraños

**Comment by Winfidonarleyan** [comment]
Is this really about cross BG?

**Comment by poszer** [comment]
Interesting. Will be great if someone else test this in game, and take a look on these chats. This doesn't mean, this issue is caused by cfbg. If i get some free time today, i'll give a test

**Comment by FrancescoBorzi** [comment]
Just to be safe: waiting for one more feedback to confirm that there are no issues here

**Comment by solidmaxtor** [comment]
Compile and test today, this is mi report

OK - Work Cross in repart players 
OK - Work correctly cross factions
ERROR - BG Chat - Not work propperly
OK - Ender BG Restor factions correct
ERROR - Ender BG not restore MORF players need reload.

testCFBG

**Comment by FrancescoBorzi** [comment]
@Winfidonarleyan please check the report from @solidmaxtor

**Comment by milestorme** [comment]
I’ve been super busy and had a lot of back log but I can test this tomorrow too on a clean staging server

**Comment by Winfidonarleyan** [comment]
Morf change at leave bg, chat need test and maybe rework, now no time

**Comment by andoys** [comment]
Testing this on my server starting today :)

**Comment by poszer** [comment]
After applying https://github.com/azerothcore/azerothcore-wotlk/pull/2064/files + module, we have a lot bugs now in battlegrounds

Issues at this moment, keep im mind that everything is not tested, so the list will be updated when i confirm new issue:

1. Random Spam Chat in Raid and Guild (If you write something in Raid chat, all of these messages will be shown also in guild, but in other structure, like numbers, symbols, etc etc - Too much spam.) 

2. If you queue bg as a group there's chance you cannot attack each other bcoz you are friendly with the opssite factions.

3. When the Finishing Battleground, players are staying in their fake morphs provided for CFBG, leaving it doesn't restore Morph players (need to relog always)

4. BG Chat not working properly (Same like first)
Everything caused by module and this PR : #2064 and module https://github.com/Winfidonarleyan/mod-cfbg

**Comment by talamortis** [comment]
Can I please add, Why are you adding more hooks to the core and altering the code when its not needed?

**Comment by Winfidonarleyan** [comment]
Okay, all full rework, go test

**Comment by andoys** [comment]
I have tested latest:

Text for CFBG Random Battleground (Levels: 80 - 80). Registered x/x announcement is not GLOBAL. Only self can see the message.


if tested with 10 accounts. Lowered Random BG queue. But queue still won't start even 10/10
crossfactionqueue

Chat spam fixed
Tauren Warrior became Human Warrior in BG but can't understand Alliance Chat. (/say /battleground)
Draenei Warrior became Blood Efl Warrior in BG bug can't understand Horde Chat (/say /battleground)

**Comment by Winfidonarleyan** [comment]
@andoys Thanks for the feedback

**Comment by Winfidonarleyan** [comment]
Okay, I update module
- [x] Fix chat for not same-origin factions https://github.com/Winfidonarleyan/mod-cfbg/commit/ed5bd361d64d1ac6842fccaa6abb74277048827e
- [x] Added support global announce (if enabled in conf) https://github.com/Winfidonarleyan/mod-cfbg/commit/0fbaf3b3d1aa4dd7999b5ac593d1092bdd22a389

**Comment by andoys** [comment]
Latest CFBG Module test:
* [x] Queue global announcement now working
bgqueue1

* [] Random BG queue still not work ? set to 5 minimum team and still won't queue up even with 10/10
bgqueue

* [x] Tauren Warrior who became Human Warrior can now be understood by all factions:
WoWScrnShot_080419_192150

* [x] Morphing also fixed after BG:
bgqueuemorph

**Comment by Winfidonarleyan** [comment]
You not set for big bg, Therefore, your queue does not work

**Comment by andoys** [comment]
Ahh its not part of this pull. 
anyway tested working

**Comment by tronx-antares** [comment]
When will it be compatible?

**Comment by Winfidonarleyan** [comment]
After merge @tronx-antares

**Comment by tronx-antares** [comment]
Oks I'll be waiting for the module because I want to try it !! @Winfidonarleyan

**Pull request merged**